.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+# This is a basic CI pipeline, which will build eradiate and
+# run the test suite on it.
+name: Eradiate build and test
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 0 * * *
+
+jobs:
+  build-and-test:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+          token: ${{ secrets.ERADIATECI_PAT }}
+          ref: public
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: eradiate
+      - name: Build and run tests
+        shell: bash -l {0}
+        run: |          
+          sudo apt-get update && DEBIAN_FRONTEND=noninteractive sudo apt-get install -y   \
+          git                                             \
+          ninja-build                                     \
+          clang-9                                         \
+          libc++-9-dev                                    \
+          libc++abi-9-dev                                 \
+          libpng-dev                                      \
+          zlib1g-dev                                      \
+          build-essential                                 \
+          libjpeg-dev                                     \
+          ninja-build                                     \
+          cmake
+          
+          make conda-init
+          
+          pip install --editable . --no-deps
+          
+          export CC=clang-9
+          export CXX=clang++-9
+          
+          source setpath.sh
+          
+          cmake --preset default
+          cmake --build build -j 10
+          
+          pytest ./tests
```

Comment:
As a separate build job impossible? It would be nice to get a visual report in case build fails, prior to inspecting the logs.